### PR TITLE
Merge changes from staging to master

### PR DESCRIPTION
**Pull request by maurelian**
This merge simply reverts the latest one, then 'unsquashes' and replays the changes that had been made. This was requested by @janaka-steph in #71. 

It does muddy up the commit history, but proper attribution in commits and blames seems like a worthwhile cause.

### CODE REVIEW

File: test/humanStandardToken.js [hunk 1]
```diff
@@ -1,345 +1,204 @@
...
-      ctr = result
-      return ctr.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]})
-    }).then(function (result) {
-      assert(false, 'The preceding call should have thrown an error.')
-    }).catch((err) => {
-      assert(evmThrewError(err), 'the EVM did not throw an error or did not ' +
-                                 'throw the expected error')
-    })
+  it('approvals: attempt withdrawal from account with no allowance (should fail)', function () {
+    return expectThrow(HST.transferFrom.call(accounts[0], accounts[2], 60, {from: accounts[1]}))
```
Comment (maurelian) [inline test/humanStandardToken.js]: @Janaka-Steph I noticed that sometimes you use `return expectThrow(...`, and others (ie. line 172 below), you use `await expectThrow(...`. 

Any reason for the difference? It's a bit odd that both work.

Comment (Janaka-Steph) [reply test/humanStandardToken.js]: Good question. Mocha test is waiting for a promise. But async/await does some magic and if you don't explicitly return a promise, the value you return will automatically be wrapped in a promise. So I think it doesn't matter here if we write `await` or `return`. But not writing neither of them should lead to unexpected issues.
This thread can be interesting to read: https://stackoverflow.com/questions/35302431/async-await-implicitly-returns-promise

Comment (maurelian) [reply test/humanStandardToken.js]: Good article, thank you. 

Do you think using `return` would make more sense semantically, as `expectThrow` is not returning a promise?

Comment (Janaka-Steph) [reply test/humanStandardToken.js]: Yes you are right, it would be better with a `return`.

**Pull request merged**